CLUSTER.md

```diff
@@ -0,0 +1,102 @@
+# Cluster mode
+
+How it works:
```

Comment:
@skudasov i would like to see here `Motivation` or `Context` section where we're explaining why we even want to need to use this feature.